aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -757,7 +757,13 @@ Tensor cummaxmin_backward(const Tensor& grad, const Tensor& input, const Tensor&
   if (input.numel() == 0) {
     return input;
   }
-  auto result = at::zeros(input.sizes(), input.options());
+  auto result = grad.new_zeros(input.sizes(), input.options());
+
+  // for composite compliance, use out-of-place variant of
+  // `scatter_add` if `indices` is a Tensor Subclass.
+  if (areAnyTensorSubclassLike({indices})) {
+    return result.scatter_add(dim, indices, grad);
+  }
```

Comment:
nit: This will be a little more performant like the following:
```
# A
auto result = at::zeros(input.sizes(), input.options());

if (areAnyTensorSubclassLike({indices, grad})) {
  # B
  return result.scatter_add(dim, indices, grad);
}
```
Consider the case where grad is a BatchedTensor. Then (A) produces a regular Tensor, while as written in this PR it produces a BatchedTensor. (Lets just say that the regular tensor has shape (3) while the batchedtensor has shape (N, 3)

In (B), the tensor produced by (A) may or may not be expanded to size (N, 3), so it's unclear if we actually save on any memory or not. But that's a choice that can be handled by the batching rule

Reply:
Makes sense. Addressed. Thanks!